### PR TITLE
core/fetcher: add support for `DutyAggregator`

### PR DESCRIPTION
**Pull request by xenowits**
Add support for `DutyAggregator` to `fetcher`.

category: feature
ticket: #1088

### CODE REVIEW

File: core/fetcher/fetcher.go [hunk 1]
```diff
@@ -130,14 +138,59 @@ func (f *Fetcher) fetchAttesterData(ctx context.Context, slot int64, defSet core
...
+			return core.UnsignedDataSet{}, errors.New("invalid beacon committee subscription")
+		}
+
+		res, err := eth2exp.CalculateCommitteeSubscriptionResponse(ctx, f.eth2Cl, &sub.BeaconCommitteeSubscription)
+		if err != nil {
+			return core.UnsignedDataSet{}, err
+		}
+
+		if !res.IsAggregator {
+			continue
```
Comment (corverroos) [inline core/fetcher/fetcher.go:161]: add comment maybe: `// This validator isn't an aggregator for this slot`

File: core/fetcher/fetcher.go [hunk 2]
```diff
@@ -130,14 +138,59 @@ func (f *Fetcher) fetchAttesterData(ctx context.Context, slot int64, defSet core
...
+			return core.UnsignedDataSet{}, errors.New("invalid attestation")
+		}
+
+		// Query BN for aggregate attestation.
+		aggAtt, err := f.eth2Cl.AggregateAttestation(ctx, eth2p0.Slot(slot), att.Data.BeaconBlockRoot)
+		if err != nil {
+			return core.UnsignedDataSet{}, err
+		}
+
+		resp[pubkey] = core.AggregatedAttestation{
```
Comment (corverroos) [inline core/fetcher/fetcher.go:181]: Isn't there a `core.NewAggregateAttesation(aggAtt)` function?

File: core/fetcher/fetcher_test.go [hunk 3]
```diff
@@ -252,3 +344,33 @@ func assertRandaoBlindedBlock(t *testing.T, randao eth2p0.BLSSignature, block co
 		require.Fail(t, "invalid block")
 	}
 }
+
+// randomSignedBeaconCommitteeSubscription returns a SignedBeaconCommitteeSubscription with the inputs and a random slot signature.
+func randomSignedBeaconCommitteeSubscription(vIdx, slot, commIdx int) core.SignedBeaconCommitteeSubscription {
```
Comment (corverroos) [inline core/fetcher/fetcher_test.go:349]: can move this to `testutil/random.go`

**Review by corverroos — APPROVED** [review]

File: core/fetcher/fetcher.go (hunk 2 above)
Comment (xenowits) [reply core/fetcher/fetcher.go:181]: No. I created a new type for aggregate attestation implementing unsignedData
```go
// AggregatedAttestation wraps Attestation and implements the UnsignedData interface.
type AggregatedAttestation struct {
	eth2p0.Attestation
}
```

**Pull request merged**